documentation/how-to/authorization.md

```diff
@@ -0,0 +1,262 @@
+temporal-admin-k8s/0*  active     idle   10.1.232.71
+temporal-k8s/0*        active     idle   10.1.232.64         auth enabled
+
+Relation provider                 Requirer                       Interface          Type     Message
+openfga-k8s:openfga               temporal-k8s:openfga           openfga            regular
+openfga-k8s:peer                  openfga-k8s:peer               openfga-peer       peer
+postgresql-k8s:database           openfga-k8s:database           postgresql_client  regular
+postgresql-k8s:database           temporal-k8s:db                postgresql_client  regular
+postgresql-k8s:database           temporal-k8s:visibility        postgresql_client  regular
+postgresql-k8s:database-peers     postgresql-k8s:database-peers  postgresql_peers   peer
+postgresql-k8s:restart            postgresql-k8s:restart         rolling_op         peer
+temporal-admin-k8s:admin          temporal-k8s:admin             temporal           regular
+temporal-k8s:peer                 temporal-k8s:peer              temporal           peer
+```
+
+To view your OpenFGA store's information, you can run the command below. Make
+note of the store's information, as this will enable us to connect to it through
+a client to add the necessary tuples in the following section.
+
+```bash
+juju show-unit temporal-k8s/0
+
+# Output:
+temporal-k8s/0:
+  opened-ports: []
+  charm: local:jammy/temporal-k8s-6
+  leader: true
+  life: alive
+  relation-info:
+    ...
+      openfga: '{"store_id": "<store_id>", "token": "<token>",
+        "address": "10.152.183.144", "port": "8080", "scheme": "http", "auth_model_id": "<model_id>"}'
+```
+
+At this point, our Temporal Server is active again with authorization enabled.
+This means that any request made to the Temporal Server will need to be
+[authenticted](./authentication.md) using Google OAuth.
+
+## Create OpenFGA Authorization Rules
+
+Once authorization is enabled, you will not be able to access any namespace in
+Temporal. You must now create the necessary tuples in the OpenFGA store to
+enable users read/write access to different namespaces.
+
+The authorization model used is as follows:
+
+```
+model
+  schema 1.1
+
+type user
+type group
+  relations
+    define member: [user]
+type namespace
+  relations
+    define admin: [group#member]
+    define reader: [group#member] or writer
+    define writer: [group#member]
+```
+
+The above model is a simple way of relating users to groups to namespaces. A
+`user` can be related to a `group` as a `member`, and a `group` can be related
+to a `namespace` as either a `reader`, `writer` or `admin`.For example, If user
+`john` is a member of group `abc`, and group `abc` is related to namespace
+`example` as a `writer`, then user `john` will be assigned write access on
+namespace `example`.
+
+To add tuples to our OpenFGA store, we can use an
+[OFGA client](https://github.com/canonical/ofga) as follows:
```

Comment:
It would be nice to have a generic ofga CLI client Snap. I can propose that within our team if it would be useful for you?

Reply:
It's definitely not ideal giving someone a block of code. But in fact, I think I recall the OpenFGA mentioning that they were working on a CLI.

Reply:
Found it! https://github.com/openfga/cli

Reply:
Thanks! I've re-factored this PR to also include charm actions that allow the user to interact with the OpenFGA store by adding, removing and checking authorization rules.